dnf-behave-tests/features/repoquery/main.feature

```diff
@@ -0,0 +1,521 @@
+ When I execute dnf with args "repoquery --deplist --latest-limit 1 top-a"
+ Then the exit code is 0
+  And stdout is
+      """
+      package: top-a-2:2.0-2.src
+
+      package: top-a-2:2.0-2.x86_64
+        dependency: bottom-a1 = 1:1.0-1
+         provider: bottom-a1-1:1.0-1.noarch
+        dependency: mid-a1 >= 2
+         provider: mid-a1-1:1.0-1.x86_64
+      """
+
+Scenario: deplist --latest-limit (deplist is an alias for repoquery --deplist)
+ When I execute dnf with args "deplist --latest-limit 1 top-a"
+ Then the exit code is 0
+  And stdout is
+      """
+      package: top-a-2:2.0-2.src
+
+      package: top-a-2:2.0-2.x86_64
+        dependency: bottom-a1 = 1:1.0-1
+         provider: bottom-a1-1:1.0-1.noarch
+        dependency: mid-a1 >= 2
+         provider: mid-a1-1:1.0-1.x86_64
+      """
+
+
+# --extras: installed pkgs, not from known repos
+Scenario: repoquery --extras
+Given I successfully execute rpm with args "-i --nodeps {context.dnf.fixturesdir}/repos/miscellaneous/x86_64/dummy-1.0-1.x86_64.rpm"
+ When I execute dnf with args "repoquery --extras"
+ Then the exit code is 0
+  And stdout is
+      """
+      dummy-1:1.0-1.x86_64
+      """
+
+Scenario: repoquery --extras (no such packages)
+ When I execute dnf with args "repoquery --extras"
+ Then the exit code is 0
+  And stdout is empty
+
+Scenario: repoquery --extras NAME (package is installed)
+Given I successfully execute rpm with args "-i --nodeps {context.dnf.fixturesdir}/repos/miscellaneous/x86_64/dummy-1.0-1.x86_64.rpm"
+Given I successfully execute rpm with args "-i --nodeps {context.dnf.fixturesdir}/repos/miscellaneous/x86_64/weird-1.0-1.x86_64.rpm"
+ When I execute dnf with args "repoquery --extras dummy"
+ Then the exit code is 0
+  And stdout is
+      """
+      dummy-1:1.0-1.x86_64
+      """
+
+Scenario: repoquery --extras NAME (package is not installed)
+ When I execute dnf with args "repoquery --extras dummy"
+ Then the exit code is 0
+  And stdout is empty
+
+
+# --installed: list only installed packages
+Scenario: repoquery --installed
+Given I successfully execute dnf with args "install bottom-a1"
+ When I execute dnf with args "repoquery --installed"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom-a1-1:2.0-1.noarch
+      """
+
+Scenario: repoquery --installed (no such packages)
+ When I execute dnf with args "repoquery --installed"
+ Then the exit code is 0
+  And stdout is empty
+
+Scenario: repoquery --installed NAME
+Given I successfully execute dnf with args "install bottom-a1 bottom-a2"
+ When I execute dnf with args "repoquery --installed bottom-a1"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom-a1-1:2.0-1.noarch
+      """
+
+Scenario: repoquery --installed NAME (no such packages)
+ When I execute dnf with args "repoquery --installed bottom-a1"
+ Then the exit code is 0
+  And stdout is empty
+
+
+# --location
+@bz1639827
+Scenario: repoquery --location NAME
```

Comment:
With http repo the test will have an additional value (using baseurl or even better with metalink).